data_structures/src/superblock.rs

```diff
@@ -345,34 +422,132 @@ impl SuperBlockState {
+                let num_total_votes = self.votes_mempool.valid_votes_counter();
+                let invalid_votes = self.votes_mempool.invalid_votes_counter();
+                let num_missing_votes = signing_committee_length - num_total_votes - invalid_votes;
+                if two_thirds_consensus(
+                    most_voted_num_votes + num_missing_votes,
+                    signing_committee_length,
+                ) || self.missing_rescue_votes(None)
+                {
+                    // There is no consensus, but if the missing votes vote the same as the
+                    // majority, there can be consensus
+                    SuperBlockConsensus::Unknown
+                } else {
+                    // There is no consensus, regardless of the missing votes
+                    SuperBlockConsensus::NoConsensus
+                }
             }
         }
     }
 
+    fn get_rescue_consensus(&self) -> Option<Hash> {
+        match self.rescue_votes_mempool.most_voted_superblock() {
+            Some((rescue_hash, counter)) => {
+                if two_thirds_consensus(counter, self.rescue_committee.len()) {
+                    // R1
+                    Some(rescue_hash)
+                } else {
+                    // R2
+                    None
+                }
+            }
+            None => {
+                // R2
+                None
+            }
+        }
+    }
+
+    fn missing_rescue_votes(&self, normal_consensus: Option<Hash>) -> bool {
+        let rescue_committee_length = self.rescue_committee.len();
+        let (_, most_voted_num_votes) = self
+            .rescue_votes_mempool
+            .most_voted_superblock()
+            .unwrap_or_default();
+        let num_total_votes = self.rescue_votes_mempool.valid_votes_counter();
+        let invalid_votes = self.rescue_votes_mempool.invalid_votes_counter();
+        let num_missing_votes = rescue_committee_length - num_total_votes - invalid_votes;
+
+        let two_thirds_consensus_normal =
+            two_thirds_consensus(most_voted_num_votes, rescue_committee_length);
+
+        let two_thirds_consensus_with_missing_votes =
+            if let Some(normal_consensus) = normal_consensus {
+                let votes_in_consensus_with_normal = self
+                    .rescue_votes_mempool
+                    .votes_counter_from_superblock(&normal_consensus);
+                two_thirds_consensus(
+                    votes_in_consensus_with_normal + num_missing_votes,
+                    rescue_committee_length,
+                )
+            } else {
+                two_thirds_consensus(
+                    most_voted_num_votes + num_missing_votes,
+                    rescue_committee_length,
+                )
+            };
+
+        matches!(
+            (
+                two_thirds_consensus_normal,
+                two_thirds_consensus_with_missing_votes,
+            ),
+            (false, true)
+        )
```

Comment:
Clippy should detect this and suggest:

```suggestion
        !two_thirds_consensus_normal && two_thirds_consensus_with_missing_votes
```